eppo_core/src/events/event_delivery.rs

```diff
@@ -0,0 +1,60 @@
+use crate::events::event::Event;
+use crate::Error;
+use log::info;
+use reqwest::{Method, StatusCode};
+use serde::{Deserialize, Serialize};
+
+pub struct EventDelivery {
+    sdk_key: String,
+    ingestion_url: String,
+    client: reqwest::Client,
+}
+
+#[derive(serde::Deserialize)]
+struct EventDeliveryResponse {
+    failed_events: Vec<String>,
```

Comment:
I believe we want `Uuid` here, so we can match it back to events we just sent out. (As a bonus, `Uuid` is more efficient than `String` for copying and comparison.)
```suggestion
    failed_events: Vec<uuid::Uuid>,
```